verilog/tools/preprocessor/verilog_preprocessor.cc

```diff
@@ -49,53 +51,172 @@ static absl::Status StripComments(const SubcommandArgsRange& args,
+  for (lexer.DoNextToken(); !lexer.GetLastToken().isEOF();
+       lexer.DoNextToken()) {
+    // For now we will store the syntax tree tokens only, ignoring all the
+    // white-space characters. however that should be stored to output the
+    // source code just like it was.
+    if (verilog::VerilogLexer::KeepSyntaxTreeTokens(lexer.GetLastToken()))
+      lexed_sequence.push_back(lexer.GetLastToken());
   }
 
-  const std::string usage = absl::StrCat("usage: ", argv[0],
-                                         " command args...\n"
-                                         "available commands:\n",
-                                         commands.ListCommands());
+  // Control flow tree constructing.
+  verilog::FlowTree control_flow_tree(lexed_sequence);
+  auto status =
+      control_flow_tree
+          .GenerateControlFlowTree();  // construct the control flow tree.
+  status = control_flow_tree.DepthFirstSearch(
+      0);  // traverse the tree by dfs from the root (node 0).
+
+  // Printing the token streams of every possible variant.
+  int variants_counter = 1;
+  for (const auto& current_variant : control_flow_tree.variants_) {
+    outs << "Variant number " << variants_counter++ << ":\n";
+    for (auto token : current_variant) outs << token << '\n';
+    puts("");
+  }
+
+  return absl::OkStatus();
+}
+
+ABSL_FLAG(bool, strip_comments, false,
+          "Replaces comments with white-spaces in files passed.");
+ABSL_FLAG(bool, multiple_cu, true,
+          "Files are preprocessed in separate compilation units.");
+ABSL_FLAG(bool, generate_variants, false,
+          "Generates every possible variants w.r.t. compiler conditionals");
+
+int main(int argc, char* argv[]) {
+  const std::string usage =
+      absl::StrCat("usage:\n", argv[0], " [options] file [<file>...]\n\n",
+                   "options summary:\n",
+                   "-multiple_cu: Files are preprocessed in separate "
+                   "compilation units.\n",
+                   "-strip_comments: Replaces one/multi-line comments with "
+                   "equal white-spaces.\n",
+                   "-generate_variants: Generates every possible variants "
+                   "w.r.t. compiler conditionals.\n");
 
   // Process invocation args.
   const auto args = verible::InitCommandLine(usage, &argc, &argv);
-  if (args.size() == 1) {
+  if (args.empty()) {
     std::cerr << absl::ProgramUsageMessage() << std::endl;
     return 1;
   }
-  // args[0] is the program name
-  // args[1] is the subcommand
-  // subcommand args start at [2]
-  const SubcommandArgsRange command_args(args.cbegin() + 2, args.cend());
-
-  const auto& sub = commands.GetSubcommandEntry(args[1]);
-  // Run the subcommand.
-  const auto status = sub.main(command_args, std::cin, std::cout, std::cerr);
-  if (!status.ok()) {
-    std::cerr << status.message() << std::endl;
+
+  // Get flags.
+  bool strip_comments_flag = absl::GetFlag(FLAGS_strip_comments);
```

Comment:
make the values assigned to `const`.

Also in other places. By default, make every variable you assign to and that is not modified later `const`; this will help readers of the code as they have to keep track of less things that are potentially mutable.